tests/testthat/test-ScpModel-Class.R

```diff
@@ -17,24 +17,30 @@ test_that("ScpModel", {
 ## Internal function that creates a minimal SE object as expected by
 ## scplainer for unit testing ScpModel class methods
 ## @param nr Number of rows
-## @param nc Number of columns
-.createMinimalData <- function(nr = 10, nc = 5) {
+.createMinimalData <- function(nr = 10) {
     require("SummarizedExperiment")
-    a <- matrix(1, nr, nc)
-    rownames(a) <- letters[1:nr]
-    colnames(a) <- LETTERS[1:nc]
+    nc <- nr * nr  # Set the number of columns to be sufficient for the largest row
+    a <- matrix(NA, nr, nc)
+    for (i in 1:nr) {
+        a[i, 1:(i * i)] <- 1:(i * i)
+    }
+    rownames(a) <- paste0("row", 1:nr)
+    colnames(a) <- paste0("col", 1:nc)
     se <- SummarizedExperiment(assays = List(assay = a))
     list(se = se, a = a)
 }
-
 ## Internal function that creates a mock List of ScpModelFit objects
 ## for unit testing ScpModel class methods
 ## @param model An ScpModel object
 ## @param features A character() with the names of the features for
 ##     which to create mock ScpModelFit objects
-.addScpModelFitList <- function(model, features) {
+## @param coef_p A logical indicating whether to create coefficients for
+## the purpose of having a p for test
+.addScpModelFitList <- function(model, features, coef_p = FALSE) {
     fitList <- as(lapply(1L:length(features), function(i) {
-        ScpModelFit(n = i * i, p = i)
+        smf <- ScpModelFit()
+        if (coef_p) scpModelFitCoefficients(smf) <- rep(0, i)
```

Comment:
it's quite counter-intuitive with best coding practices, but best practice for unit testing is that your test should depend at least as possible on internal functions, eg you can simply do: 
```r
if (coef_p) smf@coefficients <- rep(0, i)
```
Otherwise, your tests are not compartmentalised. In some cases, you cannot do otherwise, but here it's avoidable. 

Also, convention in scp (at least that's my intention), is to name variables with use `camelCase`, not 
`snake_case`